### PR TITLE
Separating Code Generation Step 6: Build Manager Service - Handling Build Status Succeed

### PR DESCRIPTION
**Pull request by EugeneTseitlin**
Closes: #3088

## PR Details

Unpack archive with generated code to FS, send ready status.

## PR Checklist
- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/code_of_conduct.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-build-manager-service/.env [hunk 1]
```diff
@@ -6,6 +6,7 @@ GENERATE_RESOURCE_TOPIC=build.internal.generate-resource.event.0
 BUILD_STATUS_TOPIC=build.internal.build-status.event.0
 
 BUILD_CONTEXT_FS_LOCATION=/home/eugene/build-contexts
```
Comment (arielweinberger) [inline packages/amplication-build-manager-service/.env:8]: Make sure to remove `/eugene` mentions in the `.env` file

File: packages/amplication-build-manager-service/src/codeBuild/codeBuild.service.ts [hunk 2]
```diff
@@ -11,49 +11,115 @@ import {
...
-import { CODE_BUILD_PROJECT_NAME } from 'src/constants';
+import { join } from 'path';
+import {
+  BUILD_ARTIFACT_S3_BUCKET,
+  BUILD_ARTIFACT_S3_LOCATION,
+  BUILD_IMAGE_NAME,
+  BUILD_IMAGE_VERSION,
+  CODE_BUILD_PROJECT_NAME,
+  GET_BUILD_BY_RUN_ID_TOPIC,
+} from 'src/constants';
```
Comment (arielweinberger) [inline packages/amplication-build-manager-service/src/codeBuild/codeBuild.service.ts:22]: Remove `src/` imports everywhere please. I have already refactored the entire codebase to get rid of them.

File: packages/amplication-build-manager-service/src/codeBuild/codeBuild.service.ts [hunk 3]
```diff
@@ -11,49 +11,115 @@ import {
...
     } catch (err) {
       throw new Error(
-        `Failed to trigger CodeBuild job run. Input: contextArchivePath: ${contextArchivePath}. Source error: ${err}`,
+        `Failed to trigger CodeBuild job run. Input: contextArchivePath: ${contextPath}. Source error: ${err}`,
       );
     }
   }
 
+  mapCodeBuildStatusToAmplicationBuildStatus(buildStatus: string): BuildStatus {
+    switch (buildStatus) {
```
Comment (arielweinberger) [inline packages/amplication-build-manager-service/src/codeBuild/codeBuild.service.ts:100]: Let's introduce an enum - `CodeBuildStatus`

File: packages/amplication-build-manager-service/src/queue/queue.service.ts [hunk 4]
```diff
@@ -27,4 +45,12 @@ export class QueueService {
     const response = this.kafkaClient.emit(topic, message);
     response.subscribe(observer);
   }
+
+  sendMessage(topic: string, message: string) {
```
Comment (arielweinberger) [inline packages/amplication-build-manager-service/src/queue/queue.service.ts:49]: I believe we should rename this service to `kafkaService`.

The term `queue` is no longer valid when this service consists of emitting events and sending request-response messages. This is simply not a capability of a queueing system.

**Review by arielweinberger — CHANGES_REQUESTED** [review]
Really well done man. A few things that need to be addressed but I'm glad we're moving so quickly with this.

File: packages/amplication-build-manager-service/src/codeBuild/codeBuild.service.ts (hunk 2 above)
Comment (EugeneTseitlin) [reply packages/amplication-build-manager-service/src/codeBuild/codeBuild.service.ts:22]: Done

File: packages/amplication-build-manager-service/src/codeBuild/codeBuild.service.ts (hunk 3 above)
Comment (EugeneTseitlin) [reply packages/amplication-build-manager-service/src/codeBuild/codeBuild.service.ts:100]: Done

File: packages/amplication-build-manager-service/src/codeBuild/codeBuild.service.ts [hunk 5]
```diff
@@ -11,49 +11,118 @@ import {
...
+      sourceLocationOverride: contextPath,
+      artifactsOverride: {
+        type: 'S3',
+        location: this.buildArtifactS3Bucket,
+        path: artifactPath,
+        name: 'artifact.zip',
+        namespaceType: 'NONE',
+        packaging: 'ZIP',
+      },
+      imageOverride: `${this.buildImageName}:${this.buildImageVersion}`,
```
Comment (EugeneTseitlin) [inline packages/amplication-build-manager-service/src/codeBuild/codeBuild.service.ts:85]: generate image override in constructor once

File: packages/amplication-build-manager-service/src/codeBuild/codeBuild.service.ts [hunk 6]
```diff
@@ -11,49 +11,118 @@ import {
...
     const command = new StartBuildCommand(input);
 
     try {
-      return this.codeBuildClient.send(command);
+      const response = await this.codeBuildClient.send(command);
+      return response.build.arn;
     } catch (err) {
       throw new Error(
-        `Failed to trigger CodeBuild job run. Input: contextArchivePath: ${contextArchivePath}. Source error: ${err}`,
+        `Failed to trigger CodeBuild job run. Input: contextArchivePath: ${contextPath}. Source error: ${err}`,
```
Comment (EugeneTseitlin) [inline packages/amplication-build-manager-service/src/codeBuild/codeBuild.service.ts:95]: chain error

File: packages/amplication-build-manager-service/src/queue/queue.service.ts [hunk 7]
```diff
@@ -27,4 +45,12 @@ export class QueueService {
...
+    return new Promise((resolve) => {
+      this.kafkaClient.send(topic, message).subscribe((response) => {
+        resolve(response);
```
Comment (EugeneTseitlin) [inline packages/amplication-build-manager-service/src/queue/queue.service.ts:52]: add observer and handle error

File: packages/amplication-build-manager-service/.env (hunk 1 above)
Comment (EugeneTseitlin) [reply packages/amplication-build-manager-service/.env:8]: Done

File: packages/amplication-build-manager-service/src/codeBuild/codeBuild.service.ts (hunk 5 above)
Comment (EugeneTseitlin) [reply packages/amplication-build-manager-service/src/codeBuild/codeBuild.service.ts:85]: done

File: packages/amplication-build-manager-service/src/codeBuild/codeBuild.service.ts (hunk 6 above)
Comment (EugeneTseitlin) [reply packages/amplication-build-manager-service/src/codeBuild/codeBuild.service.ts:98]: Done

File: packages/amplication-build-manager-service/src/queue/queue.service.ts (hunk 4 above)
Comment (EugeneTseitlin) [reply packages/amplication-build-manager-service/src/queue/queue.service.ts:49]: I prefer to do it in another PR, but on the same epic.

File: packages/amplication-build-manager-service/src/queue/queue.service.ts (hunk 7 above)
Comment (EugeneTseitlin) [reply packages/amplication-build-manager-service/src/queue/queue.service.ts:52]: done

**Pull request merged**